jmespath-core/src/main/java/io/burt/jmespath/jcf/JcfAdapter.java

```diff
@@ -143,6 +154,15 @@ public Object getProperty(Object value, String name) {
   }
 
   @Override
+  public Collection<String> getPropertyNames(Object value) {
+    if (isObject(value)) {
+      return ((Map<String, Object>) value).keySet();
```

Comment:
Not sure what you mean. The argument is created either from `#createObject` in this class, or by `JsonParser` in this package. And if it isn't I think it's reasonable that a class cast exception is raised.